iterators_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,62 @@
 #include "impl.h"
 
+#include <sstream>
+#include <iterator>
+#include <algorithm>
+#include <list>
+
+
+std::string reverseWords(const std::string& input)
+{
+    if (input.empty())
+    {
+        return {};
+    }
+
+    std::istringstream inputStream {input};
+    std::ostringstream outputStream;
+    std::list<std::string> container = {};
+
+    std::copy(std::istream_iterator<std::string>(inputStream),
+              std::istream_iterator<std::string>(),
+              std::front_inserter(container));
+
+    if (container.empty())
+    {
+        return {};
+    }
+
+    auto begin = std::begin(container);
+    auto end = std::end(container);
+    std::copy(begin, std::prev(end), std::ostream_iterator<std::string>(outputStream, " "));
```

Comment:
@smay1613 , maybe a stupid question, but why std::move(algorithm) does not work in this case (does not move container content)?